sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvalPythonExec.scala

```diff
@@ -137,3 +138,19 @@ trait EvalPythonExec extends UnaryExecNode {
     }
   }
 }
+
+/**
+ * A TaskContext aware iterator.
+ *
+ * As the Python evaluation consumes the parent iterator in a separate thread,
+ * it could consume more data from the parent even after the task ends and the parent is closed.
+ * Thus, we should use ContextAwareIterator to stop consuming after the task ends.
+ */
+private[spark] class ContextAwareIterator[IN](iter: Iterator[IN], context: TaskContext)
```

Comment:
I think we can remove `private[spark]` per SPARK-16964.

Reply:
Thanks! removed.